### ISSUE TITLE
intOnly healspare operations do not check _sentinel dtype (only _sparseMap) 

### ISSUE THREAD

**Issue by jacobic** [LSSTDESC/healsparse#36]
Hi guys,

Some quick feedback on this great new functionality. I have some healpsparse maps which contain floating values, 1.0, 0.0 and hp.UNSEEN. I want to use the `xorUnion` healsparse operation

https://github.com/LSSTDESC/healsparse/blob/281bc50caa16167de3a7107309ba2a77635a78bd/healsparse/operations.py#L170

so in order to to that I cast my `_sparseMap` to dtype int. However, even with int type maps it is still possible for the numpy ufunc to fail as the sentinel type is not checked even though `intOnly == True`. 

https://github.com/LSSTDESC/healsparse/blob/281bc50caa16167de3a7107309ba2a77635a78bd/healsparse/operations.py#L244

This can be because an int map with a float sentinal passes the `intOnly` requirements of the `_sparseMap` being dtype int but will later fail because the following line results in a dtype float array.

https://github.com/LSSTDESC/healsparse/blob/281bc50caa16167de3a7107309ba2a77635a78bd/healsparse/operations.py#L281  

To summarise I change the following in my maps to ensure the operation works as expected but think some of these check could be done within the `intOnly` check as at the moment it only checks the `_sparseMap`.

```
 m._sparseMap[np.where(m._sparseMap == hp.UNSEEN)[0]] = 0
 m._sparseMap = m._sparseMap.astype(int)
 m._sentinel = 0
```

Keep up the good work! :)

Cheers,
Jacob

**Comment by erykoff** [comment]
Closed with the new `astype` method in #69 .
